iOS/Flows/Conversation/Cells/ConversationMessageCellDatasource.swift

```diff
@@ -94,21 +81,18 @@ extension ConversationMessageCellDataSource {
 
 extension ConversationMessageCellDataSource: TimeMachineCollectionViewLayoutDataSource {
 
-    func getConversation(forItemAt indexPath: IndexPath) -> Conversation? {
-        guard let item = self.itemIdentifier(for: indexPath) else { return nil }
-
-        return ChatClient.shared.channelController(for: item.channelID).conversation
-    }
-
-    func getMessage(forItemAt indexPath: IndexPath) -> Messageable? {
+    func getTimeMachineItem(forItemAt indexPath: IndexPath) -> TimeMachineLayoutItem? {
         guard let item = self.itemIdentifier(for: indexPath) else { return nil }
         let messageController = ChatClient.shared.messageController(cid: item.channelID,
                                                                     messageId: item.messageID)
 
         return messageController.message
     }
+}
+
+extension Message: TimeMachineLayoutItem {
 
-    func frontmostItemWasUpdated(for indexPath: IndexPath) {
-        // TODO: 
+    var sortValue: Double {
+        return self.createdAt.timeIntervalSinceReferenceDate
```

Comment:
TimeMachineLayoutItem need to provide a sort value so the time machine can organize them properly.